piff/optical_model.py

```diff
@@ -0,0 +1,177 @@
+
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: optical_model
+"""
+
+from __future__ import print_function
+
+import galsim
+import fitsio
+import numpy as np
+
+from .model import Model
+from .star import Star, StarFit, StarData
+
+des_pupil_template = {'obscuration': 0.301 / 0.7174,
+                      'nstruts': 4,
+                      'diam': 4.274419,  # meters
+                      # aaron plays between 19 mm thick and 50 mm thick
+                      'strut_thick': 0.050 * (1462.526 / 4010.) / 2.0, # conversion factor is nebulous?!
+                      'strut_angle': 45 * galsim.degrees}
+
+class Optical(Model):
+    def __init__(self, rzero=0.1, sigma=0., g1=0., g2=0., pupil_path=None, lam=500., optical_template='des', logger=None, **kwargs):
+        """Initialize the Optical Model
+
+        :param rzero:               Atmospheric seeing. Usually in the 0.1 - 0.2 range. [default: 0.1]
+        :param g1, g2:              Shear to apply to final image. Simulates vibrational modes. [default: 0]
+        :param sigma:               Convolve with gaussian of size sigma. [default: 0]
+        :param pupil_path:          If a path is given, load up a pupil image, else
+                                    make image from galsim parameters referencing optical_template [default: None]
+                                    Note: if a pupil_path is specified, then a diameter needs to also be specified
+                                    in the optical_template
+        :param lam:                 Wavelength of observations in nanometers [default: 500]
+        :param optical_template:    If no pupil plane image is given, create one from a set of templates. [default: 'des']
```

Comment:
I don't like having this default to 'des'.  I think that should be required to be explicit (so it defaults to None).  Then either `pupil_path` or `optical_template` would be required.

Also, I think we want to keep `diam` as a kwarg.  If the user is not requesting to use a preset template, then they need to be able to specify `diam` directly as an argument.  So it should default to None, but check that it is given if `optical_template is None`.

Finally, to make it easier to add more templates for other cameras, I would suggest having the global (module-level) variable be optical_templates, and have it be a dict indexed by the allowed values of the `optical_template` kwarg.

```
optical_templates = { 'des' : {'obscuration' : 0.301 / 0.7174, ... } }
```

Then an lsst module could have a line that would look like
```
piff.optical_model.optical_templates['lsst'] = { 'diam' : 6.5, 'obscuration' : 0.6, ... } # or whatever these are for LSST
```
And here in the init, you would just check if `optical_template` is a valid key in the `optical_templates` dict and use `optical_templates[optical_template]`.  It's a bit more maintainable this way than the way you currently do it with an explicit check for `if optical_template == 'des'`.